src/pages/instances/actions/snapshots/CreateSnapshotForm.tsx

```diff
@@ -34,68 +28,38 @@ const CreateSnapshotForm: FC<Props> = ({ instance, close, onSuccess }) => {
   const queryClient = useQueryClient();
   const controllerState = useState<AbortController | null>(null);
 
-  const isRunning = instance.status === "Running";
-  const isStateful = instance.config["migration.stateful"];
-
-  const getStatefulInfo = () => {
-    if (isStateful && isRunning) {
-      return "";
-    }
-    if (isStateful) {
-      return `To create a stateful snapshot,\nthe instance must be running`;
-    }
-    return (
-      <>
-        {`To create a stateful snapshot, the instance needs\n`}
-        the <code>migration.stateful</code> config set to true
-      </>
-    );
-  };
-
-  const getExpiresAt = (
-    expirationDate: string,
-    expirationTime: string | null
-  ) => {
-    expirationTime = expirationTime ?? "00:00";
-    return `${expirationDate}T${expirationTime}`;
-  };
-
-  const handleEscKey = (e: KeyboardEvent<HTMLElement>) => {
-    if (e.key === "Escape") {
-      close();
-    }
-  };
-
-  const SnapshotSchema = Yup.object().shape({
+  const SnapshotSchema: unknown = Yup.object().shape({
```

Comment:
I see you put some of the validations in the snapshot util. Can we put the whole SnapshotSchema into a shared util and reuse it for create/edit?

Reply:
I also thought that, but couldn't find a way to do it easily. Currently we're referencing the `formik` object inside the schema itself, and we're allowed to do that only because we're in the same scope where it gets defined. If you move the `SnapshotSchema` to a shared util, you'd need to pass the `formik` object to it somehow, right? And that is not allowed because it's not defined yet. As an extreme countermeasure I even tried to store the `formik` object in the state, but that triggers a chain of unpleasant effects that make the code super messy, and eventually it wasn't working anyway because it was creating the "render loop" error. So I gave up eventually.

Reply:
We shouldn't use formik in the schema. I think it is only used to access other fields. We can do that [without the formik object](https://stackoverflow.com/questions/63058945/get-the-value-of-another-field-for-validation-in-yup-schema).

Reply:
This is exactly what I needed!